src/components/navigation/index.tsx

```diff
@@ -55,12 +92,40 @@ function MainNavigation({ maxWidth }: { maxWidth: ContainerProps['maxWidth'] })
             </Link>
 
             <IconButton
+              aria-describedby={id}
               aria-label="global-settings"
+              onClick={handleClick}
               sx={{ mr: 4, color: 'secondary.main', padding: 1 }}
             >
               <LanguageIcon sx={{ fontSize: '1.5rem' }} />
             </IconButton>
-
+            <Popover
+              id={id}
+              open={open}
+              anchorEl={anchorEl}
+              onClose={handleClose}
+              anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'left',
+              }}
+            >
+              <Stack>
+                {LANGUAGE_LIST.map((lang) => (
+                  <MenuItem
+                    key={lang.code}
+                    value={lang.code}
+                    onClick={() => {
+                      handleChange({
+                        target: { value: lang.code },
+                      } as SelectChangeEvent<string>);
+                      handleLanguageMenuClose();
```

Comment:
please, notice that, you are colling `handleLanguageMenuClose()` but function is empty.
I suppose you wanted to close mune, for that you already created `handleClose`, or am I missing something?

Besides, I suppose, its possible to refactor `handleChange` function.
For example:

```
const handleChange = useCallback((code: LanguageCollection['code']) => {
    i18n
      .changeLanguage(code)
      .then(() => {
        setValueToLocalStorage(LOCAL_STORAGE_KEYS.language, code);
        setAnchorEl(null);
      })
      .catch(
      // handle error
      );
  }, []);
```

Then you can call it with `code` instaed of passing `{target: { value: lang.code }}`